neurodamus/node.py

```diff
@@ -270,6 +271,12 @@ def __init__(self, config_file, options=None):
             self._spike_populations = []
             Nd.execute("cvode = new CVode()")
             SimConfig.init(config_file, options)
+            if SimConfig.use_coreneuron:
+                CoreConfig.output_root = SimConfig.output_root
+                CoreConfig.datadir = SimConfig.coreneuron_datadir
+                # Instantiate the CoreNEURON artificial cell object which is used to fill up
+                # the empty ranks. This need to be done before the circuit is finitialized
+                CoreConfig.instantiate_artificial_cell()
```

Comment:
@iomaganaris : just have this one question: could you remind me a rational for this? Because, I thought neurodamus/neuron handles the situation when some of the ranks don't have cells i.e. empty?

Reply:
I think that's the way Neurodamus handles ranks that don't have cells. It initializes an artificial cell for these ranks using the above mechanism. If we don't have this then we get an error saying that the rank doesn't have gids/cells assigned. (I don't remember now exact error but I can try and reproduce this is you want)
You can see also: https://github.com/BlueBrain/neurodamus/commit/1f12dd6e7ee32092f3b16e9ae27d883a07b2f799 and https://github.com/BlueBrain/neurodamus/commit/a1f6b9085e03bd79d0221349a8965c341f505d05

Reply:
Ok, thanks for info. Lets forget it then!